Library/Formula/hbase.rb

```diff
@@ -4,32 +4,151 @@ class Hbase < Formula
   url "https://www.apache.org/dyn/closer.cgi?path=hbase/1.1.2/hbase-1.1.2-bin.tar.gz"
   sha256 "8ca5bf0203cef86b4a0acbba89afcd5977488ebc73eec097e93c592b16f8bede"
 
-  bottle :unneeded
-
   depends_on :java => "1.6+"
-  depends_on "hadoop"
+  depends_on "hadoop" => :optional
+  depends_on "lzo" => :recommended
+  depends_on "ant" => :build if build.with? "lzo"
+
+  resource "hadoop-lzo" do
+    url "https://github.com/cloudera/hadoop-lzo/archive/0.4.14.tar.gz"
+    sha256 "aa8ddbb8b3f9e1c4b8cc3523486acdb7841cd97c002a9f2959c5b320c7bb0e6c"
+  end
 
   def install
+    ENV.java_cache if build.with? "lzo"
     rm_f Dir["bin/*.cmd", "conf/*.cmd"]
     libexec.install %w[bin conf docs lib hbase-webapps]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
-    inreplace "#{libexec}/conf/hbase-env.sh",
-      "# export JAVA_HOME=/usr/java/jdk1.6.0/",
-      "export JAVA_HOME=\"$(/usr/libexec/java_home)\""
+    if build.with? "lzo"
+      resource("hadoop-lzo").stage do
+        # Fixed upstream: https://github.com/cloudera/hadoop-lzo/blob/master/build.xml#L235
+        inreplace "build.xml",
+                  %r{(<class name="com.hadoop.compression.lzo.LzoDecompressor" />)},
+                  "\\1\n<classpath refid=\"classpath\"/>"
+        ENV["CLASSPATH"] = Dir["#{libexec}/lib/hadoop-common-*.jar"].first
+        ENV["CFLAGS"] = "-m64"
```

Comment:
The hadoop-gpl project also says you need a 32 bit jdk to build the 32 bit version: ` export JAVA_HOME=/path/to/32bit/jdk export CFLAGS=-m32 export CXXFLAGS=-m32 ant compile-native` and such a thing doesn't exist for os x: http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html you also need 32-bit liblzo2 installed and from my observations there are only 64 bit builds of that in homebrew (or at least the bottle is 64 bit, so it would require building from source).

I'm happy saying it depends on 64 bit osx, because I'm pretty sure at this point it does.